models/core-java/src/main/java/org/commonjava/indy/model/core/Group.java

```diff
@@ -86,17 +88,28 @@ public boolean addConstituent( final StoreKey repository )
 
         if ( constituents == null )
         {
-            constituents = new ArrayList<>();
+            constituents = new LinkedList<>();
         }
 
-        synchronized ( constituents )
+        synchronized ( monitor )
```

Comment:
Why can't you synchronize on the constituents list? And isn't synchronizing on an internal monitor object the same thing as synchronized(this), as long as nothing outside of the group is also synchronizing on the group itself?

Reply:
This is an IDE inspection warning, which this constituents list is not final and is possible to be changed from other threads.(like set method)

Reply:
I'd be okay with making that field final then, and initializing it to an empty List...then clear, etc. operations could just operate on the List instance.

Reply:
Ok, will change it.